### ISSUE TITLE
Can't do local::lib homedir install

### ISSUE THREAD

**Issue by dequis** [irssi/irssi#534]
Regression with 3376d324e1e174e2885efbb7f1f056bc2a0d2857 (refs #479, #132)

```
eval "$(perl -I$HOME/perl5/lib/perl5 -Mlocal::lib)"
./autogen.sh --prefix=$HOME/irssi --with-perl-lib=site
```

Before:

```
Building with Perl support ....... : static (in irssi binary)
Perl library directory ........... : /home/dx/perl5/lib/perl5/x86_64-linux-thread-multi
Install prefix ................... : /home/dx/irssi
```

After:

```
Building with Perl support ....... : static (in irssi binary)
Perl library directory ........... : (site default - /usr/lib/perl5/site_perl)
Install prefix ................... : /home/dx/irssi
```

Alternatively, with `./autogen.sh --prefix=$HOME/irssi --with-perl-lib=$HOME/perl5/lib/perl5/`

```
Perl library directory ........... : (other path in @INC - /home/dx/perl5/lib/perl5/x86_64-linux-thread-multi)
```

Or with `./autogen.sh --prefix=$HOME/irssi --with-perl-lib=$HOME/perl5`

```
Perl library directory ........... : (prepends to @INC with /set perl_use_lib - /home/dx/perl5/x86_64-linux-thread-multi)
```

No idea if the difference between `other path in @INC` and `prepends to @INC` is meaningful, but both of these seem to try to write to system global directories with make install:

```
Files found in blib/arch: installing files in blib/lib into architecture dependent library tree
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR: Can't create '/usr/lib/perl5/site_perl'
Do not have write permissions on '/usr/lib/perl5/site_perl'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 at -e line 1.
make[5]: *** [Makefile:825: pure_site_install] Error 13
```

**Comment by ailin-nemui** [comment]
you're hitting an unrelated bug which is some caching related issue in the configure script. `cd src/perl && make clean` then reconfigure should fix it. I'll check if I can track down that bugger

**Comment by dequis** [comment]
Nope, doesn't help, also happens with a clean clone.

```
$ git clone irssi irssi2
$ cd irssi2
$ git describe --tags
0.8.19-201-gc8630ac
$ eval "$(perl -I$HOME/perl5/lib/perl5 -Mlocal::lib)"
$ ./autogen.sh --prefix=$HOME/irssi --with-perl-lib=site
[...]
Perl library directory ........... : (site default - /usr/lib/perl5/site_perl)
```

**Comment by ailin-nemui** [comment]
just don't put --with-perl-lib=site there? that's a bug I fixed where local::lib would have overwritten it previously

**Comment by dequis** [comment]
```
$ ./autogen.sh --prefix=$HOME/irssi
[...]
Perl library directory ........... : (prepends to @INC with /set perl_use_lib - /home/dx/irssi/lib/perl5/x86_64-linux-thread-multi)
  - NOTE: This was automatically set to the same directory you gave with
  --prefix. If you want the perl libraries to install to their 'correct'
  path, you'll need to give --with-perl-lib=site option to configure.
  Anyway, installing perl to this directory should work just as well.
```

But that path is wrong.

**Comment by ailin-nemui** [comment]
I purposefully disabled any PERL_MM_OPT based prefix setting but maybe I need to revisit this for better local::lib compatibility. Manually specifying `--with-perl-lib=$HOME/perl5/lib/perl5` should also be fine at least for the time being set aside the caching issue

**Comment by ailin-nemui** [comment]
ok I rechecked, if I do not do this then --with-perl-lib=(site|vendor|perl) is always ignored and it just takes the local::lib path. so personally I don't like this behaviour and prefer the way I fixed it. what is your opinion?

**Comment by dequis** [comment]
Went ahead and merged #535

**Comment by dequis** [comment]
Oh also, thanks!
